cmake/FindSDL2.cmake

```diff
@@ -70,19 +70,22 @@
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
 
+# SDL 1.2 has SDL.h too. As such, the search paths should be restricted to
+# directories specific to SDL2.
 FIND_PATH(SDL2_INCLUDE_DIR SDL.h
   HINTS
   $ENV{SDL2DIR}
   PATH_SUFFIXES include/SDL2 include
   PATHS
-  ~/Library/Frameworks
-  /Library/Frameworks
+  ~/Library/Frameworks/SDL2.framework/Headers
+  /Library/Frameworks/SDL2.framework/Headers
   /usr/local/include/SDL2
   /usr/include/SDL2
   /sw # Fink
-  /opt/local # DarwinPorts
+  /opt/local/include/SDL2 # DarwinPorts
   /opt/csw # Blastwave
   /opt
+  NO_DEFAULT_PATH
```

Comment:
This breaks Nixpkgs/Nixos, which installs everything to separate prefixes and then adds `-isystem` and `-L` to supply them. `pkg-config` would also work, but this file doesn't use it.